### PR TITLE
feat(product-projections): add field to toggle markMatchingVariants

### LINKED ISSUE #207: Enable markMarchingVariants
### Expected behavior
Via the SDK, we need the option to set ```markMatchingVariant``` to ```true``` to enable this behaviour. (http://dev.commercetools.com/http-api-projects-products-search.html#matchingVariant)

### Actual behavior
At the moment, the parameter is not supported in the search command.

### PR DESCRIPTION
**Pull request by hisabimbola**
Fixes #207

- [ ] commit messages are commitizen-friendly
- [ ] code is unit tested
- [ ] code is integration tested
- [ ] public code is documented
- [ ] code is reviewed by at least one person
- [ ] code satisfies linter rules

### CODE REVIEW

File: src/spec/client/services/product-projections.spec.coffee [hunk 1]
```diff
@@ -65,6 +65,18 @@ describe 'ProductProjectionService', ->
...
     expect(@service.fuzzy()._queryString()).toBe 'fuzzy=true'
 
+  it 'should query for markMatchingVariants when set to true', ->
+    expect(@service.markMatchingVariants(true)._queryString()).toBe 'markMatchingVariants=true'
+
+  it 'should query for markMatchingVariants when set to false', ->
+    expect(@service.markMatchingVariants(false)._queryString()).toBe 'markMatchingVariants=false'
+
+  it 'should query for not set anything when no value is passed to markMatchingVariants', ->
+    expect(@service.markMatchingVariants()._queryString()).toBe ''
```
Comment (butenkor) [inline src/spec/client/services/product-projections.spec.coffee]: should it not default to ```'markMatchingVariants=false'```?

File: src/spec/client/services/product-projections.spec.coffee [hunk 2]
```diff
@@ -98,10 +110,11 @@ describe 'ProductProjectionService', ->
       .filter('foo:bar')
       .filterByQuery('foo:bar')
       .filterByFacets('foo:bar')
+      .markMatchingVariants(true)
```
Comment (butenkor) [inline src/spec/client/services/product-projections.spec.coffee]: Query for the test above does not use search [endpoint](https://github.com/sphereio/sphere-node-sdk/blob/ae72a3718dfda20adb45c91407eba7a7b653fea1/src/coffee/services/product-projections.coffee#L306) but markMatchingVariants is exclusive feature of search which is not available over product-projections endpoint.

File: src/coffee/services/product-projections.coffee [hunk 3]
```diff
@@ -65,6 +65,22 @@ class ProductProjectionService extends BaseService
...
+  # markMatchingVariants - {Boolean} `true` to do a mark product variants, `false` not mark matching product variants
+  #
+  # Returns a chained instance of `this` class
+  #
+  # Examples
+  #
+  #   service = client.productProjections
+  #   service.text('Sapphure', 'en').markMatchingVariants(true).search()
+  markMatchingVariants: (value) ->
+    return this unless _.isBoolean(value)
```
Comment (Siilwyn) [inline src/coffee/services/product-projections.coffee]: Could you explain this?

File: src/spec/client/services/product-projections.spec.coffee (hunk 1 above)
Comment (hisabimbola) [inline src/spec/client/services/product-projections.spec.coffee]: I did not add any default to the `markMatchingVariants` method, when it's called you need to explicitly pass a boolean to specify whether it should be false or not.

File: src/spec/client/services/product-projections.spec.coffee (hunk 2 above)
Comment (hisabimbola) [inline src/spec/client/services/product-projections.spec.coffee]: Calling `asSearch()` adds the search endpoint

File: src/coffee/services/product-projections.coffee (hunk 3 above)
Comment (hisabimbola) [inline src/coffee/services/product-projections.coffee]: Returns `this` if the value passed in is not a `value`. This enforces the usage of just `true` or `false`

Comment (Siilwyn) [inline src/coffee/services/product-projections.coffee]: I stil don't get why you would return `this`.

Comment (butenkor) [inline src/coffee/services/product-projections.coffee]: I guess it is required if you want to chain function calls like that: ```service.markMatchingVariants(true).limit(10).search()```

Comment (hisabimbola) [inline src/coffee/services/product-projections.coffee]: Consistency and chaining

File: src/spec/client/services/product-projections.spec.coffee (hunk 1 above)
Comment (butenkor) [inline src/spec/client/services/product-projections.spec.coffee]: Maybe @yanns knows if it should be disabled by default?

File: src/spec/client/services/product-projections.spec.coffee (hunk 2 above)
Comment (butenkor) [inline src/spec/client/services/product-projections.spec.coffee]: I just wanted to point out that only when you do ```asSearch``` only than it makes sense to set markMatchingVariants.

File: src/spec/client/services/product-projections.spec.coffee (hunk 1 above)
Comment (yanns) [inline src/spec/client/services/product-projections.spec.coffee]: @agourlay can you take care of that?

Comment (agourlay) [inline src/spec/client/services/product-projections.spec.coffee]: At the API level ```markMatchingVariants``` is enabled by default.

As we want to disable it by default in the future at the API, I believe it is a good idea to start the change from the clients.

Therefore I vote for disabling by default in the SDK if possible in order to push actively the migration.

Check this for some context https://sphere.atlassian.net/wiki/display/S/Search+matching+variants+disabled+by+default

File: src/coffee/services/product-projections.coffee (hunk 3 above)
Comment (Siilwyn) [inline src/coffee/services/product-projections.coffee]: So if you give it invalid input it just continues?

File: src/spec/client/services/product-projections.spec.coffee (hunk 1 above)
Comment (hisabimbola) [inline src/spec/client/services/product-projections.spec.coffee]: Updated and set `markMatchingVariants` to false by default. d9ec17d

**Comment by hisabimbola** [comment]
Closing the PR, this feature will be implemented in the new SDK. You can track progress [here](https://github.com/commercetools/nodejs/issues/100)

**Pull request closed without merging**